cobra/model_building/forward_selection.py

```diff
@@ -230,13 +234,19 @@ def _forward_selection(self, train_data: pd.DataFrame,
                                           .union(set(model.predictors)))
 
                 fitted_models.append(model)
+            # else:
+            #     # If model returns None for the first time,
+            #     # one can in theory stop the feature selection process
+            #     # but we leave it run such that tqdm cleanly finishes
+            #     break
 
         if not fitted_models:
-            log.error("No models found in forward selection")
+            log.error("No models found in forward selection.")
 
         return fitted_models
 
-    def _find_next_best_model(self, train_data: pd.DataFrame,
+    def _find_next_best_model(self,
+                              train_data: pd.DataFrame,
```

Comment:
just calling it "basetable" or "basetable_parts" might be better, "train_data" could suggest to people - when they're not scrolling down - that only train data is used for feature selection.

Reply:
Agree, and also thought changing it, but instead I emphasized in the documentation of `ForwardFeatureSelection.fit()` for `train_data` what the argument should be, and added an assertion to make sure it includes both a train & selection split.

Reply:
Ok.